tests/asset_canister_test.rs

```diff
@@ -0,0 +1,97 @@
+use candid::{Encode, Principal};
+use helpers::{
+    create_canister_with_cycles, get_asset_canister_wasm, retry_async, start_ic_gateway,
+    stop_ic_gateway, upload_asset_to_asset_canister, verify_canister_asset,
+};
+use hex::encode;
+use pocket_ic::PocketIcBuilder;
+use reqwest::Client;
+use std::{net::SocketAddr, str::FromStr, time::Duration};
+use tokio::runtime::Runtime;
+use tracing::info;
+mod helpers;
+
+const IC_GATEWAY_DOMAIN: &str = "gateway.icp";
+const IC_GATEWAY_ADDR: &str = "127.0.0.1:8080";
+const CANISTER_INITIAL_CYCLES: u128 = 100_000_000_000_000;
+const FETCH_ASSET_RETRY_TIMEOUT: Duration = Duration::from_secs(50);
+const FETCH_ASSET_RETRY_INTERVAL: Duration = Duration::from_secs(10);
+
+fn init_logging() {
+    tracing_subscriber::fmt()
+        .with_max_level(tracing::Level::INFO)
+        .with_test_writer()
+        .try_init()
+        .expect("failed to init logger")
+}
+
+// Test scenario:
+// - start pocket-ic server with one nns subnet
+// - start ic-gateway service connected to the pocket-ic endpoint
+// - deploy asset canister via pocket-ic interface
+// - upload some sample asset to the canister
+// - retrieve asset via HTTP client through ic-gateway endpoint
+// - verify asset integrity
+
+#[test]
+fn asset_canister_test() {
```

Comment:
Let's maybe put the actual test in some other function and call it from here? And leave the init stuff here. This way we can add more test funcs later & don't need to worry about init.

Reply:
All tests would have to go sequential for now anyway I guess, since Cargo doesn't allow us to do some preparation work before running the tests.